src/validate.ts

```diff
@@ -258,6 +286,35 @@ export function validate(
             "Appsync does not support concurrent integration invocation or methods on the `Promise` api."
           ),
         ];
+      } else if (checker.isIntegrationNode(node.expression)) {
+        /**
+         * If this is an integration node, app sync does not support integrations outside of the main function body.
+         *
+         * We check to see if the root statement (the one in the function body), if deterministic
+         * variable - const v = func()
+         * expression - await func()
+         * return - return await func()
+         *
+         * Also ensure that we are not inside an inline conditional statement.
+         *
+         * const v = x ? await func() : await func2()
+         */
+        if (
+          rootStatement &&
+          (!checker.isDeterministicStatement(rootStatement) ||
```

Comment:
Move this guard here. I don't think it belongs in the checker and it's too specific to appsync. Also, probably wouldn't call it deterministic.